src/operation/watch/mod.rs

```diff
@@ -0,0 +1,73 @@
+#[cfg(test)]
+mod test;
+use super::aggregate::{Aggregate, AggregateTarget};
+use crate::{
+    bson::{doc, Document},
+    change_stream::{ChangeStreamSpecification, ChangeStreamTarget},
+    cmap::{Command, CommandResponse, StreamDescription},
+    error::Result,
+    operation::{append_options, Operation},
+    options::ChangeStreamOptions,
+};
+#[derive(Debug)]
+pub(crate) struct Watch {
+    target: ChangeStreamTarget,
+    pipeline: Vec<Document>,
+    options: Option<ChangeStreamOptions>,
+    aggregate_operation: Aggregate,
+}
+impl Watch {
+    pub(crate) fn new(
+        target: impl Into<ChangeStreamTarget>,
+        pipeline: impl IntoIterator<Item = Document>,
+        options: Option<ChangeStreamOptions>,
+    ) -> Result<Self> {
+        let target: ChangeStreamTarget = target.into();
+        let pipeline: Vec<Document> = pipeline.into_iter().collect();
+
+        let mut options_doc: Document = Document::new();
+        append_options(&mut options_doc, options.as_ref())?;
+        let aggregate_target = match &target {
```

Comment:
Instead of defining `target`, I think this can just be `let aggregate_target = match target.into() { ... };`

Reply:
I missed that `target` is used on line 49; marking this as resolved